loader_hub/file/unstructured/base.py

```diff
@@ -0,0 +1,35 @@
+"""Unstructured file reader.
+
+A parser for unstructured text files using Unstructured.io.
+Supports .txt, .docx, .pptx, .jpg, .png, .eml, .html, and .pdf documents.
+
+"""
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+from gpt_index.readers.base import BaseReader
+from gpt_index.readers.schema.base import Document
+
+
+class UnstructuredReader(BaseReader):
+    """General unstructured text reader for a variety of files."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Init params."""
+        super().__init__(*args, **kwargs)
+
+        # Prerequisite for Unstructured.io to work
+        import nltk
+
+        nltk.download("averaged_perceptron_tagger")
+
+    def load_data(
+        self, file: Path, extra_info: Optional[Dict] = None
+    ) -> List[Document]:
+        """Parse file."""
+        from unstructured.partition.auto import partition
+
+        elements = partition(str(file))
+        text = "\n\n".join([" ".join(str(el).split()) for el in elements])
```

Comment:
there could be an option to combine partitions into one Document or separate Documents - with the latter, we can basically use unstructured capabilities for smart text splitting

Reply:
actually we could clarify this in the README too, as unstructured helps to solve some of the pain points with naive text splitting that gpt index currently implies. By default gpt index splits a Document only if it needs to (if it's above the specified chunk size). By "pre-splitting into multiple docs here, we basically offer users an alternative, smarter form of text splitting

Reply:
Makes sense

Reply:
one simple approach to experiment with could be to create a new document for each `unstructured.documents.elements.Title` and include everything underneath until the next Title. obviously not a perfect solution, and would depend on the html quite a bit.

Reply:
at the very least though, one should rarely have to split an individual text element -- the exception being a really long `.NarrativeText` element.

Reply:
Interesting, I'll note this as a todo to experiment with